### ISSUE TITLE
Group foo is not available when the group comes from ldap via nslcd

### ISSUE THREAD

**Issue by cchildress** [saltstack/salt#39018]
### Description of Issue/Question
The file.directory state is unable to set group ownership on a directory if the group is managed via LDAP instead of a local group.

### Setup
<pre>/some/path:
  file.directory:
    - user: root
    - group: foo
    - mode: '0755'</pre>

### Steps to Reproduce Issue
1. Set up minion to auth via LDAP using nslcd.
2. Use salt to create a directory that is group owned by a group in LDAP.
3. You should see group not available as the failure reason in Salt.

### Versions Report
From the master:
<pre>Salt Version:
           Salt: 2016.11.0
 
Dependency Versions:
           cffi: 1.9.1
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec  6 2016, 08:42:55)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 14.4.1
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: 4.1.6
 
System Versions:
           dist: gentoo 2.3 
        machine: x86_64
        release: 4.4.39-gentoo
         system: Linux
        version: Gentoo Base System 2.3</pre>

From the minion:
<pre>    Salt Version:
               Salt: 2016.11.0
     
    Dependency Versions:
               cffi: 1.9.1
           cherrypy: Not Installed
           dateutil: Not Installed
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.8
            libgit2: Not Installed
            libnacl: Not Installed
           M2Crypto: Not Installed
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.4.8
       mysql-python: Not Installed
          pycparser: 2.14
           pycrypto: 2.6.1
             pygit2: Not Installed
             Python: 2.7.12 (default, Dec  6 2016, 08:42:55)
       python-gnupg: Not Installed
             PyYAML: 3.12
              PyZMQ: 14.4.1
               RAET: Not Installed
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.4.2
                ZMQ: 4.1.4
     
    System Versions:
               dist: gentoo 2.3 
            machine: x86_64
            release: 4.4.39-gentoo
             system: Linux
            version: Gentoo Base System 2.3</pre>

**Comment by Ch3LL** [comment]
@cchildress looking at the code it does a check to see if the group exists and calls [this](https://github.com/saltstack/salt/blob/v2016.11.1/salt/modules/file.py#L169)

So i'm guessing the issue lies in that pyton module grp not accurately grabbing the gid of that group.

If you run the following in a python interpreter what do you see:

```
>>> import grp
>>> group = 'ldapgroup'
>>> grp.getgrnam(group).gr_gid
```

(replace ldapgroup with the name of your ldap group)

**Comment by cchildress** [comment]
@Ch3LL Yup, the grp module appears to be working as expected:
<pre>Python 2.7.12 (default, Dec  6 2016, 08:42:55) 
[GCC 4.9.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import grp
>>> group = 'git'
>>> grp.getgrnam(group).gr_gid
2007</pre>

That gid lines up with what I expect and what shows up in `getent`.

**Comment by cchildress** [comment]
It's not stale, I'm not sure why this is listed as blocked to be honest.
I got the info requested added to the issue same day. Just waiting on someone to triage this.

**Comment by rallytime** [comment]
@Ch3LL Can you come back to this one when you have a moment?

**Comment by Ch3LL** [comment]
thanks for bumping this for me @cchildress and @rallytime 

so i looked at previous issues and i found this: https://github.com/saltstack/salt/issues/12803

i'm guessing this is a duplicate and it looks like ldap authentication needs to be added to work with teh file module.

**Comment by cchildress** [comment]
Remove stale.

**Comment by cchildress** [comment]
Remove stale.

**Comment by Ch3LL** [comment]
@cchildress did you see my last comment, this is a duplicate of #12803. are you okay closing here in favor of that issue?

**Comment by cchildress** [comment]
Yeah, that sounds good to me.
